src/LieTheory/WeylGroup.jl

```diff
@@ -753,6 +753,42 @@ function reflection(beta::RootSpaceElem)
   return W([list_of_indices; current_index; reverse(list_of_indices)])
 end
 
+###############################################################################
+# G-set functionality
+
+function gset(G::WeylGroup; closed::Bool=false, check::Bool=true)
+  return GSetByElements(G, *, roots(root_system(G)); closed=closed, check=check)
+end
+
+function action_homomorphism(Omega::GSetByElements{WeylGroup,S}) where {S}
+  W = acting_group(Omega) # our base group
+
+  # Compute a permutation group `G` isomorphic with `W`.
+  phi = isomorphism(PermGroup, W)
+  G = codomain(phi) # permutation group
+
+  # Let `G` act on `Omega` as `W` does.
+  phiinv = inv(phi)
+  actfun = action_function(Omega)
+  fun = function (omega::S, g::PermGroupElem)
+    return actfun(omega, phiinv(g))
+  end
+
+  OmegaG = GSetByElements(G, fun, Omega; closed=true, check=false)
+
+  # Compute the permutation action on `1:length(Omega)`
+  # corresponding to the action of `W` on `Omega`.
+  return compose(phi, action_homomorphism(OmegaG))
+end
+
+# Currently implemented to allow gset computations that require image.
+function image(phi::Generic.CompositeMap{WeylGroup,PermGroup})
+  return sub(codomain(phi), [image(phi, x) for x in gens(domain(phi))])
+end
+
+# Required for blocks computation.
+on_sets(set::T, x::WeylGroupElem) where {T<:AbstractSet} = T(pnt * x for pnt in set)
```

Comment:
It would be better to instead widen the signature of https://github.com/oscar-system/Oscar.jl/blob/0de68cf20a4a01face94b5cf152d5d03ae092f23/src/Groups/action.jl#L136 and the similar function around (at least all that have a pure julia implementation and no calls to GAP)

Reply:
But that uses `pnt^x` not `pnt*x`, does that work the `WeylGroupElem` cases we care about?

Reply:
True. But couldn't we just add a `^` method for WeylGroupElem that behaves the same as `*`? There was no real reason behind the decision which function to use for their action, so I am fine to add this (in a non-breaking way)

Reply:
The meaning of the `on_sets` method mentioned above is:
For a set `set` and a GAPGroupElem `x`, the generic `on_sets(set, x)` assumes that `^` for an element of `set` and `x` is the correct action, and we compute a set of images w.r.t. this action.

If we want `WeylGroupElem` to behave like this, what is the right solution:
Generalize the signatures in `action.jl` to `Union{GAPGroupElem, WeylGroupElem}`, introduce a new intermediate type between `GAPGroupElem` and `GroupElem` that is also a supertype of `WeylGroupElem`, or try to get the behaviour for `GroupElem`?